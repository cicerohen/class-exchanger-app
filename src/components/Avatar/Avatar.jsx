import React from 'react';
import PropTypes from 'prop-types';
import Style from './style/Avatar.scss';

const Avatar = ({ src, style }) => (
  <div className={style.root}>
    <img src={src} className={Style.image} alt="" />
  </div>
);

Avatar.propTypes = {
  src: PropTypes.string.isRequired,
  style: PropTypes.objectOf(PropTypes.any),
};

Avatar.defaultProps = {
  style: Style,
};

export default Avatar;
