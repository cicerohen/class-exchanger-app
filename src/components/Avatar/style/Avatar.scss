@import './Avatar.mixins';

.root {
  display: flex;
  width: 150px;
  img {
    border-color: #fff;
    border-size: 2px;
    border-radius: 50%;
    max-width: 100%;
  } 
}
