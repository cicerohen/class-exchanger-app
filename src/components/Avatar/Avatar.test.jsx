import React from 'react';
import { shallow } from 'enzyme';
import Avatar from '../Avatar';

const minimalProps = {
  src: 'https://avatars3.githubusercontent.com/u/819041',
};

const commonProps = {
  ...minimalProps,
};

describe('<Avatar /> Tests', () => {
  it('should render without exploding', (done) => {
    const wrapper = shallow(<Avatar {...minimalProps} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', (done) => {
    const wrapper = shallow(<Avatar {...minimalProps} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should render correctly with all props', (done) => {
    const wrapper = shallow(<Avatar {...commonProps} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
