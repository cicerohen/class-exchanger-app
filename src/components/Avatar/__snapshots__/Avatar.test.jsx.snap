// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Tests should render correctly 1`] = `
<div
  className="root"
>
  <img
    alt=""
    className="image"
    src="https://avatars3.githubusercontent.com/u/819041"
  />
</div>
`;

exports[`<Avatar /> Tests should render correctly with all props 1`] = `
<div
  className="root"
>
  <img
    alt=""
    className="image"
    src="https://avatars3.githubusercontent.com/u/819041"
  />
</div>
`;
